mqtt_client updates

brunnels opened the pull request:
Also updated the vcproj to remove some cruft

brunnels committed:
mqtt_client updates

brunnels committed:
Merge branch 'master' of https://github.com/jeelabs/esp-link

brunnels committed:
Added buildtime wifi config ability
Added SDK_DBG parameter to turn off nonsense SDK debug output

tve commented on esp-link/main.c:
```diff
@@ -79,7 +79,19 @@ static void ICACHE_FLASH_ATTR prHeapTimerCb(void *arg) {
+char* esp_link_version = VERS_STR(VERSION);
+
 void user_rf_pre_init(void) {
+  system_set_os_print(DEBUG_SDK);
```
what is the purpose of this?

brunnels replied:
this turns off the default esp sdk debugging statements if it's DEBUG_SDK is not defined.

tve commented on esp-link/main.c:
```diff
@@ -79,7 +79,19 @@ static void ICACHE_FLASH_ATTR prHeapTimerCb(void *arg) {
+  os_strncpy((char*)stconf.password, VERS_STR(STA_PASS), 64);
+  stconf.bssid_set = 0;
+  wifi_station_set_config_current(&stconf);
+#endif
```
This creates an inconsistency between what's in the config flash and what the radio is configured with. What does the Wifi web page and the home page show? Also, making a change on the Wifi web page essentially has no effect, which is rather confusing.

brunnels replied:
Other option could be to get the value from the flash wifi config and if it's not the same as the values in STA_PASS or STA_SSID then write the new values to flash.  This would remove any ambiguity.

tve commented on esp-link/mqtt_client.c:
```diff
@@ -3,91 +3,137 @@
+      os_strcpy(statusTopicStr, flashConfig.mqtt_clientid);
+      os_strcat(statusTopicStr, "/");
+      os_strcat(statusTopicStr, flashConfig.mqtt_status_topic);
+    }
```
this creates an inconsistency between what the user sees on the web page and what is actually used, and while you may like the clientid to be prepended that's not necessarily other's preference.

brunnels replied:
Okay.  If I can just remove that buildup of the topic with the client_id or if you agree, I can add a GUI tickbox for "prepend client id" and have the client_id shown on the webpage when it's ticked.

tve commented on include/espmissingincludes.h:
```diff
@@ -36,9 +36,21 @@ void ets_timer_setfn(ETSTimer *t, ETSTimerFunc *fn, void *parg);
+#undef os_printf
+#define os_printf(format, ...) \
+  system_set_os_print(true); \
+  os_printf_plus(format, ## __VA_ARGS__); \
```
what is os_printf_plus ?

brunnels replied:
os_printf_plus is what os_printf is actually macro'd to in the SDK.

brunnels committed:
Changed mqtt statusTopicStr to not pre-pend client_id

brunnels committed:
Merge branch 'master' of https://github.com/jeelabs/esp-link

brunnels committed:
Added ifdef checks for mqtt enabled
Added better handling of compile time config of SSID and PASSWORD
Removed default disable of ESP SDK debug log messages

brunnels committed:
Missing define

brunnels committed:
Missing statement
